src/state/project-state.ts

```diff
@@ -1,53 +1,53 @@
-namespace App {
-  type Listener<T> = (items: T[]) => void;
-  class State<T> {
-    protected listeners: Listener<T>[] = [];
-    addListener(listenerFn: Listener<T>) {
-      this.listeners.push(listenerFn);
-    }
+import { Project } from "../models/project.js";
+import { ProjectStatus } from "../models/project.js";
+type Listener<T> = (items: T[]) => void;
+class State<T> {
+  protected listeners: Listener<T>[] = [];
+  //array of listeners where we store the functions that need to be called whenever the state changes or we add a project
```

Comment:
great comment, if you can rename the property to something that speaks for itself it will be even greater. So if instead of "listeners" we call it "stateChangeListeners" it will be obvious to most that this is a array of listeners that store functions...  and this will make the comment more or less not needed.